src/app/app-routing.ts

```diff
@@ -235,15 +251,12 @@ const routes: Routes = [
   },
   /** For use when donor clicks logout in the menu on the wordpress site **/
   {
-    component: LoginComponent, // Angular requires we set a component but it will never be used as `canActivate` always
-                               // redirects.
+    component: LoginComponent, // Angular requires we set a component but it will never be used client-side as      
+                                    // `canActivate` always redirects there.
     path: 'logout',
     pathMatch: 'full',
     canActivate: [
-      async () => {
-        inject(IdentityService).logout();
-        await inject(Router).navigate(['/'], {replaceUrl: true});
```

Comment:
I replaced direct navigation with a URL tree return, having read at https://github.com/angular/angular/issues/56691#issuecomment-2191394819 that
> Navigating inside a guard is not recommended. you should return an URL tree of the boolean.